src/main/java/de/tum/in/www1/artemis/service/connectors/AbstractGitService.java

```diff
@@ -0,0 +1,337 @@
+     * @param remoteRepositoryUri The URI of the remote repository, not null.
+     * @param defaultBranch       The name of the default branch to be checked out, not null.
+     * @param isBare              Whether the repository is a bare repository (without working directory)
+     * @return The configured Repository instance.
+     * @throws IOException             If an I/O error occurs during repository initialization or configuration.
+     * @throws InvalidRefNameException If the provided default branch name is invalid.
+     *
+     *                                     <p>
+     *                                     The method disables automatic garbage collection of the repository to prevent potential issues with file deletion
+     *                                     as discussed in multiple resources. It also avoids the use of symbolic links for security reasons,
+     *                                     following best practices against remote code execution vulnerabilities.
+     *                                     </p>
+     *
+     *                                     <p>
+     *                                     References:
+     *                                     <ul>
+     *                                     <li>https://stackoverflow.com/questions/45266021/java-jgit-files-delete-fails-to-delete-a-file-but-file-delete-succeeds</li>
+     *                                     <li>https://git-scm.com/docs/git-gc</li>
+     *                                     <li>https://www.eclipse.org/lists/jgit-dev/msg03734.html</li>
+     *                                     </ul>
+     *                                     </p>
+     */
+    @NotNull
+    public static Repository linkRepositoryForExistingGit(Path localPath, VcsRepositoryUri remoteRepositoryUri, String defaultBranch, boolean isBare)
+            throws IOException, InvalidRefNameException {
+        // Create the JGit repository object
+
+        // Open the repository from the filesystem
+        FileRepositoryBuilder builder = new FileRepositoryBuilder();
+
+        if (isBare) {
+            builder.setBare();
+        }
+        // bare repositories use a different path than checked out ones
+        builder.setGitDir(isBare ? localPath.toFile() : localPath.resolve(".git").toFile()).setInitialBranch(defaultBranch).setMustExist(true).readEnvironment().findGitDir()
+                .setup(); // scan environment GIT_* variables
+
+        try (Repository repository = new Repository(builder, localPath, remoteRepositoryUri)) {
+            // disable auto garbage collection because it can lead to problems (especially with deleting local repositories)
+            // see https://stackoverflow.com/questions/45266021/java-jgit-files-delete-fails-to-delete-a-file-but-file-delete-succeeds
+            // and https://git-scm.com/docs/git-gc for an explanation of the parameter
+            // and https://www.eclipse.org/lists/jgit-dev/msg03734.html
```

Comment:
These links are already in the javadoc of this method. Do we want to repeat them again?